### PR TITLE
Create 611802 - SQLConnection.ConnectionString throws NRE

### PR DESCRIPTION
**Pull request by saurabh500**
Documenting known issue for .Net Framework System.Data NRE.

### CODE REVIEW

File: releases/net472/KnownIssues/611802 - SQLConnection.ConnectionString throws NullReferenceException .md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+The code change sets the internal value of retry count without validating if the ConnectionString is a non-null or non-empty value
+which results in the NullReferenceException.
+
+## Impact
+
+The applications using SqlConnection.ConnectionString API to set an empty connection string will receive an exception.
+
+## Workarounds
+
+No Known workarounds yet with .Net 4.7.2.
```
Comment (vivmishra) [inline releases/net472/KnownIssues/611802 - SQLConnection.ConnectionString throws NullReferenceException .md]: No known workarounds yet with .NET 4.7.2.

File: releases/net472/KnownIssues/611802 - SQLConnection.ConnectionString throws NullReferenceException .md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+## Cause
+
+.NET Framework 4.7.2 includes changes for increasing the default retry count to Azure SQL DB to 2. 
+The code change sets the internal value of retry count without validating if the ConnectionString is a non-null or non-empty value
+which results in the NullReferenceException.
+
+## Impact
+
+The applications using SqlConnection.ConnectionString API to set an empty connection string will receive an exception.
```
Comment (vivmishra) [inline releases/net472/KnownIssues/611802 - SQLConnection.ConnectionString throws NullReferenceException .md]: The impact limited to applications using SqlConnection.ConnectionString API to set an **empty connection string** that will now receive an exception.

File: releases/net472/KnownIssues/611802 - SQLConnection.ConnectionString throws NullReferenceException .md (hunk 1 above)
Comment (saurabh500) [reply releases/net472/KnownIssues/611802 - SQLConnection.ConnectionString throws NullReferenceException .md]: Should I remove the "yet" from this? It seems to imply that we could have a workaround, but thats not the case until the bug fix is delivered?

File: releases/net472/KnownIssues/611802 - SQLConnection.ConnectionString throws NullReferenceException .md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+## Workarounds
+
+No Known workarounds yet with .Net 4.7.2.
+
+## Resolution
+
+Product team working on code fix.
```
Comment (vivmishra) [inline releases/net472/KnownIssues/611802 - SQLConnection.ConnectionString throws NullReferenceException .md]: .NET team is working on a fix that will be available in a future update.  

_(we sho0uld update this section when the update is out)_

File: releases/net472/KnownIssues/611802 - SQLConnection.ConnectionString throws NullReferenceException .md [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# NullReferenceException while setting `SqlConnection.ConnectionString` to an empty string or null.
+## Symptoms
+
+When an application tries to set the `ConnectionString` property of the `SqlConnection` object a `NullReferenceException` is thrown.
```
Comment (vivmishra) [inline releases/net472/KnownIssues/611802 - SQLConnection.ConnectionString throws NullReferenceException .md]: Add some context that this happens when you install 4.7.2.

File: releases/net472/KnownIssues/611802 - SQLConnection.ConnectionString throws NullReferenceException .md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+The code change sets the internal value of retry count without validating if the `ConnectionString` is a non-null or non-empty value
+which results in the `NullReferenceException`.
+
+## Impact
+
+The impact is limited to applications using `SqlConnection.ConnectionString` API to set an empty connection string that will now receive an exception.
+
+## Workarounds
+
+No Known workarounds with .NET 4.7.2.
```
Comment (vivmishra) [inline releases/net472/KnownIssues/611802 - SQLConnection.ConnectionString throws NullReferenceException .md]: I would say: Known --> known

File: releases/net472/KnownIssues/611802 - SQLConnection.ConnectionString throws NullReferenceException .md [hunk 6]
```diff
@@ -0,0 +1,23 @@
+# NullReferenceException while setting `SqlConnection.ConnectionString` to an empty string or null.
+## Symptoms
+
+When an application tries to set the `ConnectionString` property of the `SqlConnection` object to an empty string or null value, a `NullReferenceException` is thrown. This exceptions will occur when .NET 4.7.2 is installed on the OS hosting the application.
```
Comment (rpetrusha) [inline releases/net472/KnownIssues/611802 - SQLConnection.ConnectionString throws NullReferenceException .md]: - exceptions will occur --> exception occurs
- .NET 4.7.2 --> .NET Framework 4.7.2

File: releases/net472/KnownIssues/611802 - SQLConnection.ConnectionString throws NullReferenceException .md [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+
+## Cause
+
+.NET Framework 4.7.2 includes changes for increasing the default connection retry count, while connection to Azure SQL DB, to 2.
```
Comment (rpetrusha) [inline releases/net472/KnownIssues/611802 - SQLConnection.ConnectionString throws NullReferenceException .md]: includes changes for increasing --> increases 
count, while connection to Azure SQL DB, to 2 --> for connections to Azure SQL Database to two

File: releases/net472/KnownIssues/611802 - SQLConnection.ConnectionString throws NullReferenceException .md [hunk 8]
```diff
@@ -0,0 +1,23 @@
+# NullReferenceException while setting `SqlConnection.ConnectionString` to an empty string or null.
+## Symptoms
+
+When an application tries to set the `ConnectionString` property of the `SqlConnection` object to an empty string or null value, a `NullReferenceException` is thrown. This exceptions will occur when .NET 4.7.2 is installed on the OS hosting the application.
+
+
+## Cause
+
+.NET Framework 4.7.2 includes changes for increasing the default connection retry count, while connection to Azure SQL DB, to 2. 
+The code change sets the internal value of retry count without validating if the `ConnectionString` is a non-null or non-empty value
```
Comment (rpetrusha) [inline releases/net472/KnownIssues/611802 - SQLConnection.ConnectionString throws NullReferenceException .md]: - The code change --> This change
- of retry count --> of the retry count
- comma after value

**Comment by vivmishra** [comment]
Plz also update the PR to add the change to the index file. 
https://github.com/Microsoft/dotnet/blob/master/releases/net472/dotnet472-known-issues.md

File: releases/net472/KnownIssues/611802 - SQLConnection.ConnectionString throws NullReferenceException .md [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+which results in the `NullReferenceException`.
+
+## Impact
+
+The impact is limited to applications using `SqlConnection.ConnectionString` API to set an empty connection string that will now receive an exception.
```
Comment (rpetrusha) [inline releases/net472/KnownIssues/611802 - SQLConnection.ConnectionString throws NullReferenceException .md]: using --> using the
string that will now receive an exception --> string, which now throws an exception

File: releases/net472/KnownIssues/611802 - SQLConnection.ConnectionString throws NullReferenceException .md (hunk 5 above)
Comment (rpetrusha) [inline releases/net472/KnownIssues/611802 - SQLConnection.ConnectionString throws NullReferenceException .md]: .NET 4.7.2 --> .NET Framework 4.7.2

File: releases/net472/KnownIssues/611802 - SQLConnection.ConnectionString throws NullReferenceException .md [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+
+## Workarounds
+
+No Known workarounds with .NET 4.7.2.
+
+## Resolution
+
+.NET team is working on a fix that will be available in a future update.
```
Comment (rpetrusha) [inline releases/net472/KnownIssues/611802 - SQLConnection.ConnectionString throws NullReferenceException .md]: .NET team --> The .NET team

File: releases/net472/KnownIssues/611802 - SQLConnection.ConnectionString throws NullReferenceException .md [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+
+## Cause
+
+.NET Framework 4.7.2 includes changes for increasing the default connection retry count, while connection to Azure SQL DB, to 2. 
+The code change sets the internal value of retry count without validating if the `ConnectionString` is a non-null or non-empty value
+which results in the `NullReferenceException`.
+
+## Impact
+
+The impact is limited to applications using `SqlConnection.ConnectionString` API to set an empty connection string, that will now result in an exception.
```
Comment (rpetrusha) [inline releases/net472/KnownIssues/611802 - SQLConnection.ConnectionString throws NullReferenceException .md]: string, that will now result in an exception --> string, which now throws an exception.

File: releases/net472/KnownIssues/611802 - SQLConnection.ConnectionString throws NullReferenceException .md [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+
+## Workarounds
+
+No known workarounds with .NET 4.7.2.
```
Comment (rpetrusha) [inline releases/net472/KnownIssues/611802 - SQLConnection.ConnectionString throws NullReferenceException .md]: .NET 4.7.2 --> .NET Framework 4.7.2

File: releases/net472/KnownIssues/611802 - SQLConnection.ConnectionString throws NullReferenceException .md [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+
+## Resolution
+
+.NET team is working on a fix that will be available in a future update.
```
Comment (rpetrusha) [inline releases/net472/KnownIssues/611802 - SQLConnection.ConnectionString throws NullReferenceException .md]: .NET team --> The .NET team

**Comment by saurabh500** [comment]
@rpetrusha  I have addressed the comments. Let me know what you think.

**Pull request merged**